spigot_v1_17_R1_2/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/regen/Regen_v1_17_R1_2.java

```diff
@@ -297,17 +299,32 @@ protected void cleanup() {
 
     @Override
     protected ProtoChunk createProtoChunk(int x, int z) {
-        return new ProtoChunk(new ChunkCoordIntPair(x, z), ChunkConverter.a, freshNMSWorld) {
-            public boolean generateFlatBedrock() {
-                return generateFlatBedrock;
-            }
+        return PaperLib.isPaper()
+                ? new FastProtoChunk(new ChunkCoordIntPair(x, z), ChunkConverter.a, freshNMSWorld, freshNMSWorld) // paper
+                : new FastProtoChunk(new ChunkCoordIntPair(x, z), ChunkConverter.a, freshNMSWorld); // spigot
+    }
 
-            // no one will ever see the entities!
-            @Override
-            public List<NBTTagCompound> z() {
-                return Collections.emptyList();
-            }
-        };
+    private class FastProtoChunk extends ProtoChunk {
+
+        // avoid warning on paper
+        public FastProtoChunk(ChunkCoordIntPair pos, ChunkConverter upgradeData, LevelHeightAccessor world, WorldServer level) {
+            super(pos, upgradeData, world, level);
+        }
+
+        // compatibility with spigot
+        public FastProtoChunk(ChunkCoordIntPair pos, ChunkConverter upgradeData, LevelHeightAccessor world) {
+            super(pos, upgradeData, world);
+        }
+
+        public boolean generateFlatBedrock() {
+            return generateFlatBedrock;
+        }
+
+        // no one will ever see the entities!
```

Comment:
Why?

Reply:
That's copied over from the code before. Otherwise, the code would run into AsyncCatcher when spawning entities in the fake world while regen does not even copy them over later.